From 110d98f2bc848d16c6fdef7a486ed9478f775661 Mon Sep 17 00:00:00 2001
From: Barry Warsaw <barry@debian.org>
Date: Thu, 8 Oct 2015 08:45:16 -0700
Subject: Calculate the correct .pth path

 cov-core wants to install the file init_cov_core.pth so that cov_core_init
 will be imported and initialized automatically for subprocess coverage, if
 the environment variable COV_CORE_SOURCE is set (which e.g. nose does
 automatically).  However the upstream setup.py tries to install this file
 directly into /usr/lib/pythonX.Y/{site,dist}-packages, which clearly doesn't
 work when building the package.  This patch hard codes the .pth path for a
 location within the build directory.  dh_python{2,3} does the rest.
Forwarded: not-needed

Patch-Name: pth-installation-path.patch
---
 setup.py | 28 ++++++++--------------------
 1 file changed, 8 insertions(+), 20 deletions(-)

diff --git a/setup.py b/setup.py
index 0989863..0440d10 100644
--- a/setup.py
+++ b/setup.py
@@ -48,23 +48,11 @@ setuptools.setup(name='cov-core',
                               'Topic :: Software Development :: Testing'])
 
 if sys.argv[1] in ('install', 'develop'):
-    for path in sys.path:
-        if (path.endswith('site-packages')) or (path.endswith('dist-packages')
-                                                and 'local' in path):
-            path = os.path.join(path, PTH_FILE_NAME)
-            try:
-                pth_file = open(path, 'w')
-                pth_file.write(PTH_FILE)
-                pth_file.close()
-                sys.stdout.write('\nWrote pth file for subprocess '
-                                 'measurement to %s\n' % path)
-                break
-            except Exception:
-                sys.stdout.write('\nFailed to write pth file for subprocess '
-                                 'measurement to %s\n' % path)
-                sys.stdout.write(PTH_FILE_FAILURE)
-                break
-    else:
-        sys.stdout.write('\nFailed to find site-packages or dist-packages '
-                         'dir to put pth file in.\n')
-        sys.stdout.write(PTH_FILE_FAILURE)
+    majmin = sys.version_info[0:2]
+    pth_path = os.path.join(
+        'debian',
+        ('python3-cov-core' if majmin[0] == 3 else 'python-cov-core'),
+        'usr', 'lib', 'python{}.{}'.format(*majmin), 'dist-packages',
+        PTH_FILE_NAME)
+    with open(pth_path, 'w') as fp:
+        fp.write(PTH_FILE)
